app/process_proposal.go

```diff
@@ -79,25 +79,63 @@ func (app *App) ProcessProposal(req abci.RequestProcessProposal) abci.ResponsePr
 
 		tx, err := app.txConfig.TxDecoder()(malleatedTx.Tx)
 		if err != nil {
-			// we don't reject the block here because it is not a block validity
-			// rule that all transactions included in the block data are
-			// decodable
-			continue
+			return abci.ResponseProcessProposal{
+				Result: abci.ResponseProcessProposal_REJECT,
+			}
 		}
 
+		var checkedTx bool
+
 		for _, msg := range tx.GetMsgs() {
 			if sdk.MsgTypeURL(msg) != types.URLMsgPayForBlob {
 				continue
 			}
 
+			if !checkedTx {
+				err = tx.ValidateBasic()
+				if err != nil {
+					app.Logger().Error("Tx including MsgPayForBlob is invalid")
+					return abci.ResponseProcessProposal{
+						Result: abci.ResponseProcessProposal_REJECT,
+					}
+				}
+
+				response := app.BaseApp.CheckTx(abci.RequestCheckTx{
+					Tx:   rawTx,
+					Type: abci.CheckTxType_New,
+				})
```

Comment:
[blocking]
We can not use `CheckTx` here, as it uses its own copy of the state, not the actual state! I think its very probable that this would not be deterministic, since the check tx state depends on which txs that validator has added to their mempool.

we also only need to check the signature, and therefore also looking up the account.

Reply:
Ah makes sense, I will revert to previous changes